packages/api/Tests/UnitTests/AccessToken/AccessTokenServiceTests.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System.Collections.Generic;
+using AlvTime.Persistence.Repositories;
+using System.Linq;
+using System.Security.Claims;
+using AlvTime.Business.AccessTokens;
+using AlvTime.Persistence.DataBaseModels;
+using AlvTimeWebApi.Controllers.Utils;
+using Microsoft.AspNetCore.Http;
+using Moq;
+using Xunit;
+
+namespace Tests.UnitTests.AccessToken
+{
+    public class AccessTokenServiceTests
+    {
+        [Fact]
+        public void GetActiveAccessTokens_UserSpecified_ActiveTokensForUser()
+        {
+            var dbContext = new AlvTimeDbContextBuilder()
+                .WithUsers()
+                .CreateDbContext();
+
+            var service = CreateAccessTokenService(dbContext);
+
+            var tokens = service.GetActiveTokens();
+
+            Assert.Equal(dbContext.AccessTokens.Where(x => x.UserId == 1).ToList().Count, tokens.Count());
+        }
+
+        [Fact]
+        public void CreateLifetimeToken_FriendlyNameSpecified_TokenWithFriendlyNameCreated()
+        {
+            var dbContext = new AlvTimeDbContextBuilder()
+                .WithPersonalAccessTokens()
+                .WithUsers()
+                .CreateDbContext();
+
+            var service = CreateAccessTokenService(dbContext);
+
+            service.CreateLifeTimeToken("new token");
+
+            var tokens = service.GetActiveTokens();
+
+            Assert.Equal(dbContext.AccessTokens.Where(x => x.UserId == 1).ToList().Count(), tokens.Count());
+        }
+
+        [Fact]
+        public void DeleteToken_TokenIdSpecified_TokenWithIdDeleted()
+        {
+            var dbContext = new AlvTimeDbContextBuilder()
+                .WithPersonalAccessTokens()
+                .WithUsers()
+                .CreateDbContext();
+
+            var service = CreateAccessTokenService(dbContext);
+
+            service.DeleteActiveTokens(new List<int>{1});
+
+            var tokens = service.GetActiveTokens();
+
+            Assert.Empty(tokens);
+        }
+
+        private static AccessTokenService CreateAccessTokenService(AlvTime_dbContext dbContext)
+        {
+            var mockHttpContextAccessor = GetHttpContextAccessor();
+
+            var userContext = new UserContext(mockHttpContextAccessor.Object);
```

Comment:
For this purpose it might be easier to read the test if we mock the IUserContext to just return a given user, instead of mocking the IHttpContextAccessor?